src/player.c

```diff
@@ -1554,8 +1554,11 @@ int xmp_start_player(xmp_context opaque, int rate, int format)
 		mod->len = 0;
 	}
 
-	if (mod->len == 0 || mod->chn == 0) {
+	if (mod->len == 0) {
```

Comment:
how do this manage leading to crashes?

Reply:
The two blocks of code above this skip invalid patterns at the beginning of the module. These invalid patterns can't be scanned, so their `xxo_info` entries are default values, which includes a BPM value of 0. The function `update_from_ord_info` (called at line 1571) sets the player BPM to the `xxo_info` BPM. The 0 BPM is used to compute `s->ticksize` in mixer.c:330, which resolves to `inf` due to float division by 0. At 584 this is converted to `int`, which seems to specifically result in `INT_MIN`, and then `downmix_int_16bit` tries to iterate on that and reads/writes well past the input/output buffer bounds.

Reply:
ok.